myia/compile/backends/relay.py

```diff
@@ -523,24 +543,27 @@ def run(self, graph, context, target):
 
         for g in mng.graphs:
             if g.parent is None:
-                self.graph_map[g] = relay.GlobalVar(g.debug.debug_name)
+                if g is graph:
+                    self.graph_map[g] = relay.GlobalVar("main")
```

Comment:
This could collide with functions named "main" by the user, no?

Reply:
Yes it could.  However relay requires that the entry function is named "main" and that there is only one.  So we would have to mangle the "main" from the user somehow.

Reply:
We could probably just mangle all function names with a `myia_` prefix.